[xenmgr] Reconnect vifs on ndvm reboot

crogers1 opened the pull request:
Fix a few bugs along this control path in addition to new logic
  for applyVmBackendShift. When ndvm_x reboots, forall frontends
  with ndvm_x as a backend, cleanup all existing frontend nics and
  call into networkdaemon to hookup new vifs for ndvm_x.

  OXT-1011

Signed-off-by: Chris Rogers <rogersc@ainfosec.com>

jean-edouard commented:
Built here: http://openxt-builder.ainfosec.com:8010/builders/openxt/builds/754

jean-edouard commented:
Code LGTM.
Some changes seem purely cosmetic, but make the code more readable, so +1.

rossphilipson commented:
So with these fixes in (and probably due to this one), when you restart the NDVM it starts endlessly creating vif nodes (I had wifi off). It is still going and Windows PnP is trying its best to keep up (it is still going as I speak) but I assume it will crash at some point. So not fixed...

crogers1 commented:
@rossphilipson: I updated all the relevant PRs.  Instead of going through network daemon, I'm calling xl network-attach from xenmgr where I have all the information about the vif and can be very specific.  I also added 'devid' as a valid param to network-attach to prevent that endless vif creation loop you saw.  Worth rebuilding everything trying the test again.
